### PR TITLE
MWPW-123567: added close listener to modal and added default focus behaviour to cl…

### PR DESCRIPTION
**Pull request by pulkitSoni526**
- added a listener on the document whenever dialog id opened so that we can know when esc key is pressed and removed this listener on the close of dialog
- added delay to focus of close button so that user gets default focus on the close button of opened dialog

Resolves:  MWPW-123567(Closed)(https://jira.corp.adobe.com/browse/MWPW-123567)

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/drafts/pulkit/coca-cola-case-study-2
- After: https://MWPW-123567--milo--adobecom.hlx.page/drafts/pulkit/coca-cola-case-study-2

### CODE REVIEW

File: libs/blocks/modal/modal.js [hunk 1]
```diff
@@ -126,16 +126,20 @@ export async function getModal(el, custom) {
...
-    if (event.key === 'Escape') {
-      closeModals([dialog]);
-    }
-  });
 
   dialog.append(close, content);
   document.body.append(dialog);
   dialog.insertAdjacentElement('afterend', curtain);
-  firstFocusable.focus(focusVisible);
+  document.getElementsByClassName("dialog-close")[0]?.focus({ focusVisible: true });
```
Comment (auniverseaway) [inline libs/blocks/modal/modal.js:133]: This is not ideal due to having multiple modals on page at a given time.

Isn't the close button created in this function already?

@JasonHowellSlavin didn't this already get addressed?

**Review by auniverseaway — CHANGES_REQUESTED** [review]

File: libs/blocks/modal/modal.js [hunk 2]
```diff
@@ -126,16 +126,20 @@ export async function getModal(el, custom) {
...
+
+  const closeModal = (event) => {
+    var code = event.keyCode || event.which;
+    if (code === 27) {
```
Comment (chrischrischris) [inline libs/blocks/modal/modal.js:137]: Why the change from `event.key === 'Escape'`?

Comment (JasonHowellSlavin) [reply libs/blocks/modal/modal.js:137]: +1 . event.keyCode is deprecated: https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode

File: libs/blocks/modal/modal.js (hunk 1 above)
Comment (JasonHowellSlavin) [reply libs/blocks/modal/modal.js:133]: Yes. When I did the focus trap. Close is not the expected default focus when there is content within a modal that contains focusable items. https://www.w3.org/WAI/ARIA/apg/example-index/dialog-modal/dialog

Comment (pulkitSoni526) [reply libs/blocks/modal/modal.js:133]: @JasonHowellSlavin @auniverseaway I added default focus on close button as a part of requirement in ticket. https://jira.corp.adobe.com/browse/MWPW-123567  should this be changed  or removed from requirement as jason said it should not focus on close by default when there in content in modal.

File: libs/blocks/modal/modal.js (hunk 2 above)
Comment (pulkitSoni526) [reply libs/blocks/modal/modal.js]: removed deprecated keycode property and added your suggestion

**Comment by chrischrischris** [comment]
@pulkitSoni526 @JasonHowellSlavin @auniverseaway 
What's the status on this PR?  Are these changes still needed?

File: libs/blocks/modal/modal.js (hunk 1 above)
Comment (JasonHowellSlavin) [reply libs/blocks/modal/modal.js:133]: We clarified within comments on the ticket, so this should be removed based on WCAG/W3 a11y requirements.

**Comment by JasonHowellSlavin** [comment]
> What's the status on this PR? Are these changes still needed?

A few changes yes. But that being said, if the focus does go to the video when a video is present, the event listener for the escape key will not work because the focus will be on an iframe's document, and not the document that contains our dialog. It could be a better solution that does not require rewriting the `closeModal` function would be to check if the modal content is an iframe, and then shift focus to the close button if there is no title/input. Then again, perhaps the accessibility team should way in on the correct behavior for modals with iframes/videos since tabbing through the video controls would lead to the modal close button eventually.

File: libs/blocks/modal/modal.js [hunk 3]
```diff
@@ -112,6 +112,13 @@ export async function getModal(details, custom) {
     e.preventDefault();
   });
 
+  curtain.addEventListener('click', (e) => {
```
Comment (vhargrave) [inline libs/blocks/modal/modal.js:115]: we already have a close listener for the curtain on line 145
```
if (!dialog.classList.contains('curtain-off')) {
    const curtain = createTag('div', { class: 'modal-curtain is-open' });
    curtain.addEventListener('click', (e) => {
      if (e.target === curtain) closeModal(dialog);
    });
    dialog.insertAdjacentElement('afterend', curtain);
  }
```

File: libs/blocks/modal/modal.js [hunk 4]
```diff
@@ -120,6 +127,16 @@ export async function getModal(details, custom) {
 
   dialog.append(close);
   document.body.append(dialog);
+  dialog.insertAdjacentElement('afterend', curtain);
```
Comment (vhargrave) [inline libs/blocks/modal/modal.js:130]: the curtain was also already being added . Why is it being added again here?

File: libs/blocks/modal/modal.js [hunk 5]
```diff
@@ -120,6 +127,16 @@ export async function getModal(details, custom) {
...
+  document.getElementsByClassName("dialog-close")[0]?.focus({ focusVisible: true });
+
+  const closeModal = (event) => {
+    var keyPressed = event.key;
+    if (keyPressed?.toLowerCase() === 'escape') {
+      document.removeEventListener('keydown', closeModal);
+    }
+  }
+  document.addEventListener('keydown', closeModal);
```
Comment (vhargrave) [inline libs/blocks/modal/modal.js:139]: if you're going to add an event listener to the document, you should probably also remove the event listener on a modal close :)

**Review by vhargrave — CHANGES_REQUESTED** [review]
Unit Tests are failing and need to be fixed
Additionally, the After Link wasn't working for me. Could you get that fixed so that it's easier to test? I haven't "tested" the functionality yet. 
There are a few other things that I've found, that also need to be worked on.
